docs/help.md

```diff
@@ -0,0 +1,241 @@
+
+## (4) Invalid field <a id="4"></a>
+
+> Invalid field: The field `???` does not exist on `???`,
+> but the GraphQL document expects it to exist.<br />
+> Traversal will continue, however this may lead to undefined behavior!
+
+Similarly to the previous warning, when you're passing an introspected
+schema to the cache exchange, it is able to check whether all your queries are valid.
+This warning occurs when an unknown field is found on a selection set as part
+of a query or fragment.
+
+Check whether your schema is up-to-date or whether you're using an invalid
+field somewhere, maybe due to a typo.
+
+As the warning states, this won't lead any operation to abort or an error
+to be thrown!
+
+## (5) Invalid Abstract type <a id="5"></a>
+
+> Invalid Abstract type: The type `???` is not an Interface or Union type
+> in the defined schema, but a fragment in the GraphQL document is using it
+> as a type condition.
+
+When you're passing an introspected schema to the cache exchange, it becomes
+able to deterministically check whether an entity in the cache matches a fragment's
+type condition.
+
+This applies to full fragments (`fragment _ on Interface`) or inline fragments
+(`... on Interface`), that apply to interfaces instead of to a concrete object typename.
+
+Check whether your schema is up-to-date or whether you're using an invalid
+field somewhere, maybe due to a typo.
+
+## (6) readFragment(...) was called with an empty fragment <a id="6"></a>
+
+> readFragment(...) was called with an empty fragment.
+> You have to call it with at least one fragment in your GraphQL document.
+
+You probably have called `cache.readFragment` with a GraphQL
+document that doesn't contain a main fragment.
+
+This error occurs when no main fragment can be found, because the `DocumentNode`
+is maybe empty or does not contain fragments.
+
+When you're calling a fragment method, please ensure that you're only passing one
+or more fragments in your GraphQL document.
+
+## (7) Can't generate a key for readFragment(...) <a id="7"></a>
+
+> Can't generate a key for readFragment(...).
+> You have to pass an `id` or `_id` field or create a custom `keys` config for `???`.
+
+You probably have called `cache.readFragment` with data that the cache can't generate a
+key for.
+
+This may either happen because you're missing the `id` or `_id` field or some other
+fields for your custom `keys` config.
+
+Please make sure that you include enough properties on your data so that `readFragment`
+can generate a key.
+
+## (8) Invalid resolver data <a id="8"></a>
+
+> Invalid resolver value: The resolver at `???` returned an invalid typename that
+> could not be reconciled with the cache.
+
+This error may occur when you provide a cache resolver for a field using `resolvers` config.
+
+The value that you returns needs to contain a `__typename` field and this field must
+match the `__typename` field that exists in the cache, if any. This is because it's not
```

Comment:
Maybe we should add something on how a user can find the typename, like you can always double check this in graphql playground for example? A lot of people don't really know that this is an extra gql provides